lua/telescope/_extensions/cheat.lua

```diff
@@ -0,0 +1,86 @@
+local has_telescope, telescope = pcall(require, 'telescope')
+
+if not has_telescope then
+  error('This plugins requires nvim-telescope/telescope.nvim')
+end
+
+local curl = require("plenary.curl")
+local finders = require("telescope.finders")
+local make_entry = require("telescope.make_entry")
+local pickers = require("telescope.pickers")
+local sorters = require("telescope.sorters")
+local conf = require("telescope.config").values
+local previewers = require("telescope.previewers")
+local utils = require('telescope.utils')
+local putils = require('telescope.previewers.utils')
+local actions = require('telescope.actions')
+local defaulter = utils.make_default_callable
+local M = {}
+
+require('telescope').setup {}
+
+M.query = function(qry)
+  local q, ft
+
+  if qry == nil or vim.tbl_isempty(qry) then
+    q = ":list"
+  elseif qry.child == nil then
+    q = string.format("%s/:list", qry.ft)
+    ft = qry.ft
+  else
+    q = string.format("%s/%s", qry.ft, qry.child)
+    ft = qry.ft
+  end
+  return {
+    res = vim.split(curl.get{ url = "cht.sh/"..q.."?T" }.body or "", "\n"),
+    ft = ft or ""
+  }
+end
+
+M.get_content = function(ft, child)
+  return M.query({ft = ft, child = child})
+end
+
+M.cheat_fd = function(opts) -- TODO Make it non blocking
+  opts = opts or {}
+  local res = M.query({ft = opts.ft, child = opts.child})
+  local ft = res.ft
+  pickers.new(opts, {
+    prompt_title = 'Cheats',
+    finder = finders.new_table{ results = res.res, },
+    sorter = conf.generic_sorter(opts),
+    ft = res.ft, -- incase we want to have recusive finders.
+    previewer = previewers.new_buffer_previewer{
+      get_buffer_by_name = function(_, entry)
+        return entry.value
+      end,
+      define_preview = function(self, entry, status) -- FIXME:
+        putils.with_preview_window(status, nil, function()
+          local output = M.query{ft = self.ft, child = entry.value}
+          vim.api.nvim_buf_set_lines(self.state.bufnr, 0, -1, true, output)
+          putils.highlighter(self.state.bufnr, self.state.ft)
+        end)
+      end
+    },
+    attach_mappings = function(bufnr, map)
+      map("i", "<cr>", function(bufnr)
+        -- TODO: have float preview open instead, unless preview buffer can be
+        -- editited and copied from
```

Comment:
You can keep last bufnr and open that in a split/vsplit/tab/ that window, float.

Reply:
Oh nice, so I just pass the bufnr to float opener function. how about with vsplit, vim.cmd(....